src/ethernet_interface.cpp

```diff
@@ -931,17 +931,30 @@ void EthernetInterface::writeConfigurationFile()
         }
     }
     {
-        auto& dhcp = config.map["DHCP"].emplace_back();
-        dhcp["ClientIdentifier"].emplace_back("mac");
+        auto& dhcp4 = config.map["DHCPV4"].emplace_back();
```

Comment:
It should be [DHCPv6] as per https://www.freedesktop.org/software/systemd/man/systemd.network.html

Reply:
Updated accordingly